### PR TITLE
chore(persona): upgrade sdk to v2

### PR DESCRIPTION
**Pull request by cajubelt**
### Description

Upgrades persona SDK to v2 and updates styling. Styling still to be finalized and some styles on Android needs fixing, which can be fixed in a follow up.

Note: for testing purposes, the persona button component is included in the home screen. This isn't included in this PR

iOS:

https://user-images.githubusercontent.com/5062591/190319413-6159e8e1-5b28-4315-b1b7-b9f8258b1735.mp4

Android:

https://user-images.githubusercontent.com/5062591/190319673-a1b7eaff-a241-4953-9a2b-3e1c9a481def.mp4

### Other changes

N/A

### Tested

Manually by including the component in the home screen, unit tests

### How others should test

N/A

### Related issues

- Fixes ACT-400

### Backwards compatibility

Yes

### CODE REVIEW

File: .github/workflows/e2e-ci.yml [hunk 1]
```diff
@@ -87,8 +87,10 @@ jobs:
       # `if` conditions can't directly access secrets, so we use a workaround
       # See https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow
       SECRETS_AVAILABLE: ${{ secrets.SECRETS_AVAILABLE }}
+      BASH_ENV: /Users/administrator/.profile
     name: iOS
-    runs-on: ios-e2e-group
+    # TODO(satish): revert this once all machines are upgraded
+    runs-on: ios-e2e-group-xcode-13
```
Comment (satish-ravi) [inline .github/workflows/e2e-ci.yml:93]: These are part of #2884, just added here to confirm CI passes

File: android/app/src/main/res/drawable/rn_persona_button.xml [hunk 2]
```diff
@@ -2,23 +2,23 @@
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
   <item android:state_enabled="false">
     <shape android:shape="rectangle">
-      <corners android:radius="30dp"/>
-      <padding android:left="8dp" android:top="4dp" android:right="8dp" android:bottom="4dp"/>
+      <corners android:radius="25dp"/>
```
[the remark this answers is not in the document]
Comment (jophish) [reply android/app/src/main/res/drawable/rn_persona_button.xml:5]: (same q for the following file)

File: src/account/Persona.tsx [hunk 3]
```diff
@@ -47,24 +47,28 @@ const Persona = ({ kycStatus, text, onCancelled, onError, onPress, onSuccess }:
...
+          Logger.error(TAG, `Inquiry failed for ${inquiryId}`)
+        } else {
+          onSuccess?.()
+          ValoraAnalytics.track(CICOEvents.persona_kyc_success)
+          Logger.info(TAG, `Inquiry completed for ${inquiryId}`)
+        }
       })
-      .onCancelled(() => {
-        onCancelled?.()
+      .onCanceled(() => {
```
Comment (jophish) [inline src/account/Persona.tsx:70]: a very important breaking API change :laughing:

File: src/account/Persona.tsx [hunk 4]
```diff
@@ -47,24 +47,28 @@ const Persona = ({ kycStatus, text, onCancelled, onError, onPress, onSuccess }:
...
       .referenceId(walletAddress)
       .environment(networkConfig.personaEnvironment)
       .iosTheme(pjson.persona.iosTheme)
-      .onSuccess((inquiryId: string, attributes: InquiryAttributes) => {
-        onSuccess?.(attributes?.address)
-        ValoraAnalytics.track(CICOEvents.persona_kyc_success)
-        Logger.info(TAG, `Inquiry completed for ${inquiryId}`)
+      .onComplete((inquiryId: string, status: string, _fields: Fields) => {
+        // status can be completed or failed
+        if (status === 'failed') {
```
Comment (jophish) [inline src/account/Persona.tsx:60]: wish there was an exported enum for this :frowning_face:

**Review by jophish — APPROVED** [review]
A'ing, though can't really give any insightful comments into the changes related to our build system.

File: android/app/src/main/res/drawable/rn_persona_button.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (satish-ravi) [reply android/app/src/main/res/drawable/rn_persona_button.xml:5]: these are auto generated with the persona-tool cli per their docs. But yeah some styles don't work. Making a separate issue to address these

File: src/account/Persona.tsx (hunk 4 above)
Comment (satish-ravi) [reply src/account/Persona.tsx:60]: yeah I couldn't find one, but I'll look again one more time.

**Comment by ValoraQA** [comment]
Hey @MuckT since NA is mentioned in how other should test section ,we are skipping this PR 

CC: @cajubelt 
Thanks.!

**Pull request merged**